SimTKcommon/include/SimTKcommon/internal/Pathname.h

```diff
@@ -126,11 +128,40 @@ class SimTK_SimTKCOMMON_EXPORT Pathname {
     /// for the currently running platform (i.e. backslash for
     /// Windows and forward slash everywhere else).
     static void deconstructPathname(    const std::string& name,
-                                        bool&        isAbsolutePath,
+                                        bool&        dontApplySearchPath,
                                         std::string& directory,
                                         std::string& fileName,
                                         std::string& extension);
 
+    /// An extension of deconstructPathname(). Given a specified working directory 
+    /// (swd) and path, this function evaluates the absolute path of a given path
+    /// relative to the swd and returns the directory, fileName, and extension of 
+    /// the canonicalized path with respect to a swd, if needed. This means, for 
+    /// the path, that instead of evaluating "." as the current working directory 
+    /// (cwd), the swd is used. Unlike deconstructPathname(), this function will
+    /// always return an absolute path, and no bool dontApplySearchPath is returned.
+    /// Rules:
+    /// - If the swd is empty (after removing whitespace), deconstructPathname()
+    ///   is called, and cwd is prepended if needed to make it an absolute path.
+    /// - Otherwise, we evaluate path relative to the swd. These steps are as follows:
+    /// 1) Preprocess the swd. This means that if the swd is of any form that denotes
+    ///    an absolute path (i.e. "C:/file.ext", "C:file.ext", "./file.ext", "/file.ext")
+    ///    we change the swd to reflect the absolute path (e.g. "./file.ext" may change
+    ///    to "/cwd/file.ext" or "C:/cwdOnC/file.ext").
+    /// 2) If path is a root-relative path name (and on Windows this includes a drive) 
```

Comment:
I think (2) actually is the first rule applied. If so should come first in this list.